### DISCUSSION TITLE
Does every realtime subscription create a new client connection?

### DISCUSSION THREAD

**Discussion by miguel-JARS** [supabase/supabase#28825] [Questions] [answered]
I have a web project that creates a client connection to Supabase via a singleton. This client connections subscribes to multiple tables with realtime. Does every subscription create a new connection? Or is it only 1 connection per client.

**Comment by GaryAustin1** [comment, marked as answer]
You can have many channels on one connection.  The connection is the websocket to the realtime server and there should be one.

**Reply by caseybaggz** [reply]
@GaryAustin1 what are you referring to by "connection"? I'm assuming by channel you are meaning the `.channel` method but I'm not sure?

I have this same design and as the app is scaling am wondering if I'm supposed to have one big ass realtime data store (doesn't seem right) or stick with my many channels depending on which view you have active?

**Reply by GaryAustin1** [reply]
Connection is the websocket.  When you create a channel for the first time a websocket connection is made between the client browser and the realtime server.  Each additional channel uses the same websocket connection.  

Not sure what your data store comment means.
